### ISSUE TITLE
Better Lucee support for CFLint

### ISSUE THREAD

**Issue by TheRealAgentK** [cflint/CFLint#414]
Collection ticket for discussion around Lucee and CFLint

**Comment by TheRealAgentK** [comment]
From @KamasamaK:

I have not used Lucee extensively, so if anyone can comment on syntax differences that would need to be taken into consideration that would be great as parse errors would be the biggest issue. Currently, I know that Lucee makes semicolons optional and allows for trailing commas (#365). There is also the proposal for an [HTML island](https://dev.lucee.org/t/syntax-for-html-island-in-script/2347), but as far as I am aware nothing has come of it.

Another syntactic difference is the way Lucee uses tags in script compared to the CF11+ generic syntax. ACF requires parentheses as I believe they are implemented as functions.

**ACF**
```
cfsavecontent(variable="contentVar") {
    ...
}
```

**Lucee**
```
savecontent variable="contentVar" {
    ...
}
```

**Comment by KamasamaK** [comment]
I have not used Lucee extensively, so if anyone can comment on syntax differences that would need to be taken into consideration that would be great as parse errors would be the biggest issue. Currently, I know that Lucee makes semicolons optional and allows for trailing commas (#365). There is also the proposal for an [HTML island](https://dev.lucee.org/t/syntax-for-html-island-in-script/2347), but as far as I am aware nothing has come of it.

### Tag to script syntax
Another syntactic difference is the way Lucee uses tags in script compared to the CF11+ generic syntax. ACF requires parentheses as I believe they are implemented as functions.

**ACF**
```
cfsavecontent(variable="contentVar") {
    ...
}
```

**Lucee**
```
savecontent variable="contentVar" {
    ...
}
```

### Arrow function/Lambda syntax
**ACF**
```
g = function(x,y) {
    return x*y;
}
```

**Lucee**
```
h = (x,y) -> x*y
```

### Lucee-exclusive operators
- `===`: Check value and type
- `CT`: Alias for `CONTAINS`
- `NCT`: Alias for `DOES NOT CONTAIN`

**Comment by cybersonic** [comment]
I am also testing but a big one (that I think more developers would use) would be 
```
<cfscript>
dump(SERVER);
</cfscript>
```
There is also the ternary and elvis operators: 

 - `var item = FileExists(thing) ? "Yep" : "Nope";`
 - `var item = rockstar?:"Elvis";`


More info here: http://docs.lucee.org/guides/developing-with-lucee-server/operators.html

**Comment by michaelborn** [comment]
Yeah, what would it take to add arrow function support? Currently CFLint throws a NullPointerException when it encounters them.

(I can open this as a new issue if you'd like - I just thought I'd put it here since you mentioned it above.)

https://bitbucket.org/michaelborn_me/cfevents/addon/pipelines/home#!/results/4

**Comment by AFO-0** [comment]
Hi!  
It's seems the arrow function is not supported yet.  

And, here is another syntax from Lucee that is no well handled.  
This is the **oredered struct** definition using array brackets:  

```
var z = [
  a = 1,
  b = 2
];
```  

That is equivalent to this ugly piece of code:  

```
var z = new struct( 'ordered' );
z.a = 1;
z.b = 2;

```  

Thanks!
